``useless-else-on-loop`` should not be raised on `return`

### Bug description

Here's a typical example of W0120 being raised:

```python
def f():
    while True:
        if condition:
            pass
    else:
        pass
```

And this code **doesn't** raise W0120 (correctly):

```python
def f():
    while True:
        if condition:
            break
    else:
        pass
```

But, this code is pretty much identical in function and it **does** raise W0120 (bug):

```python
def f():
    while True:
        if condition:
            return
    else:
        pass
```

The check for W0120 should check for both `break` and `return` in the while block.

### Configuration

_No response_

### Command used

```shell
pylint mytest.py
```


### Pylint output

```shell
************* Module mytest
mytest.py:5:4: W0120: Else clause on loop without a break statement (useless-else-on-loop)
```


### Expected behavior

Issue not raised

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.6-dev0
Python 3.10.0 (default, Oct 13 2021, 06:45:00) [Clang 13.0.0 (clang-1300.0.29.3)]
```


### OS / Environment

_No response_

### Additional dependencies

_No response_

--- a/ChangeLog
+++ b/ChangeLog
@@ -37,6 +37,10 @@ Release date: TBA
 
   Closes #5281
 
+* Better warning messages for useless else or elif when a function return early.
+
+  Closes #5614
+
 * Fixed false positive ``consider-using-dict-comprehension`` when creating a dict
   using a list of tuples where key AND value vary depending on the same condition.
 

--- a/ChangeLog
+++ b/ChangeLog
@@ -37,7 +37,7 @@ Release date: TBA
 
   Closes #5281
 
-* Better warning messages for useless else or elif when a function return early.
+* Better warning messages for useless else or elif when a function returns early.
 
   Closes #5614
 

--- a/doc/whatsnew/2.13.rst
+++ b/doc/whatsnew/2.13.rst
@@ -86,6 +86,10 @@ Other Changes
 
   Closes #5648
 
+* Better warning messages for useless else or elif when a function return early.
+
+  Closes #5614
+
 * Fixed false positive ``consider-using-dict-comprehension`` when creating a dict
   using a list of tuples where key AND value vary depending on the same condition.
 

--- a/doc/whatsnew/2.13.rst
+++ b/doc/whatsnew/2.13.rst
@@ -86,7 +86,7 @@ Other Changes
 
   Closes #5648
 
-* Better warning messages for useless else or elif when a function return early.
+* Better warning messages for useless else or elif when a function returns early.
 
   Closes #5614
 

--- a/pylint/checkers/base.py
+++ b/pylint/checkers/base.py
@@ -522,7 +522,7 @@ class BasicErrorChecker(_BasicChecker):
         ),
         "W0120": (
             "Else clause on loop without a break statement, remove the else and"
-            "de-indent all the code inside it",
+            " de-indent all the code inside it",
             "useless-else-on-loop",
             "Loops should only have an else clause if they can exit early "
             "with a break statement, otherwise the statements under else "

--- a/pylint/checkers/base.py
+++ b/pylint/checkers/base.py
@@ -521,7 +521,8 @@ class BasicErrorChecker(_BasicChecker):
             "has abstract methods and is instantiated.",
         ),
         "W0120": (
-            "Else clause on loop without a break statement",
+            "Else clause on loop without a break statement, remove the else and"
+            "de-indent all the code inside it",
             "useless-else-on-loop",
             "Loops should only have an else clause if they can exit early "
             "with a break statement, otherwise the statements under else "

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -239,7 +239,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "with statement assignment and exception handler assignment.",
         ),
         "R1705": (
-            'Unnecessary "%s" after "return"',
+            'Unnecessary "%s" after "return" remove it and de-indent all the code inside it',
             "no-else-return",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a return statement. "
@@ -339,7 +339,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "Used when an if expression can be replaced with 'bool(test)'. ",
         ),
         "R1720": (
-            'Unnecessary "%s" after "raise"',
+            'Unnecessary "%s" after "raise" remove it and de-indent all the code inside it',
             "no-else-raise",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a raise statement. "
@@ -360,7 +360,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "Instead of using exit() or quit(), consider using the sys.exit().",
         ),
         "R1723": (
-            'Unnecessary "%s" after "break"',
+            'Unnecessary "%s" after "break" remove it and de-indent all the code inside it',
             "no-else-break",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a break statement. "
@@ -369,7 +369,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "break statement.",
         ),
         "R1724": (
-            'Unnecessary "%s" after "continue"',
+            'Unnecessary "%s" after "continue" remove it and de-indent all the code inside it',
             "no-else-continue",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a continue statement. "

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -239,7 +239,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "with statement assignment and exception handler assignment.",
         ),
         "R1705": (
-            'Unnecessary "%s" after "return" remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "return", remove it and de-indent all the code inside it',
             "no-else-return",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a return statement. "
@@ -339,7 +339,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "Used when an if expression can be replaced with 'bool(test)'. ",
         ),
         "R1720": (
-            'Unnecessary "%s" after "raise" remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "raise", remove it and de-indent all the code inside it',
             "no-else-raise",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a raise statement. "
@@ -360,7 +360,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "Instead of using exit() or quit(), consider using the sys.exit().",
         ),
         "R1723": (
-            'Unnecessary "%s" after "break" remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "break", remove it and de-indent all the code inside it',
             "no-else-break",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a break statement. "
@@ -369,7 +369,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "break statement.",
         ),
         "R1724": (
-            'Unnecessary "%s" after "continue" remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "continue", remove it and de-indent all the code inside it',
             "no-else-continue",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a continue statement. "

--- a/pylint/checkers/refactoring/refactoring_checker.py
+++ b/pylint/checkers/refactoring/refactoring_checker.py
@@ -239,7 +239,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "with statement assignment and exception handler assignment.",
         ),
         "R1705": (
-            'Unnecessary "%s" after "return", remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "return", %s',
             "no-else-return",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a return statement. "
@@ -339,7 +339,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "Used when an if expression can be replaced with 'bool(test)'. ",
         ),
         "R1720": (
-            'Unnecessary "%s" after "raise", remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "raise", %s',
             "no-else-raise",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a raise statement. "
@@ -360,7 +360,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "Instead of using exit() or quit(), consider using the sys.exit().",
         ),
         "R1723": (
-            'Unnecessary "%s" after "break", remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "break", %s',
             "no-else-break",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a break statement. "
@@ -369,7 +369,7 @@ class RefactoringChecker(checkers.BaseTokenChecker):
             "break statement.",
         ),
         "R1724": (
-            'Unnecessary "%s" after "continue", remove it and de-indent all the code inside it',
+            'Unnecessary "%s" after "continue", %s',
             "no-else-continue",
             "Used in order to highlight an unnecessary block of "
             "code following an if containing a continue statement. "
@@ -667,10 +667,11 @@ class RefactoringChecker(checkers.BaseTokenChecker):
 
         if _if_statement_is_always_returning(node, returning_node_class):
             orelse = node.orelse[0]
-            followed_by_elif = (orelse.lineno, orelse.col_offset) in self._elifs
-            self.add_message(
-                msg_id, node=node, args="elif" if followed_by_elif else "else"
-            )
+            if (orelse.lineno, orelse.col_offset) in self._elifs:
+                args = ("elif", 'remove the leading "el" from "elif"')
+            else:
+                args = ("else", 'remove the "else" and de-indent the code inside it')
+            self.add_message(msg_id, node=node, args=args)
 
     def _check_superfluous_else_return(self, node):
         return self._check_superfluous_else(

--- a/pylint/extensions/check_elif.py
+++ b/pylint/extensions/check_elif.py
@@ -31,7 +31,7 @@ class ElseifUsedChecker(BaseTokenChecker):
     name = "else_if_used"
     msgs = {
         "R5501": (
-            'Consider using "elif" instead of "else", replace the "else" by an "elif",'
+            'Consider using "elif" instead of "else", remove the "else",'
             ' then de-indent the code and switch from "if" to "elif" inside the else',
             "else-if-used",
             "Used when an else statement is immediately followed by "

--- a/pylint/extensions/check_elif.py
+++ b/pylint/extensions/check_elif.py
@@ -31,7 +31,8 @@ class ElseifUsedChecker(BaseTokenChecker):
     name = "else_if_used"
     msgs = {
         "R5501": (
-            'Consider using "elif" instead of "else if"',
+            'Consider using "elif" instead of "else", replace the "else" by an "elif"'
+            ' then de-indent the code and switch from "if" to "elif" inside the else',
             "else-if-used",
             "Used when an else statement is immediately followed by "
             "an if statement and does not contain statements that "

--- a/pylint/extensions/check_elif.py
+++ b/pylint/extensions/check_elif.py
@@ -31,8 +31,7 @@ class ElseifUsedChecker(BaseTokenChecker):
     name = "else_if_used"
     msgs = {
         "R5501": (
-            'Consider using "elif" instead of "else", remove the "else",'
-            ' then de-indent the code and switch from "if" to "elif" inside the else',
+            'Consider using "elif" instead of "else" then "if" to remove one indentation level',
             "else-if-used",
             "Used when an else statement is immediately followed by "
             "an if statement and does not contain statements that "

--- a/pylint/extensions/check_elif.py
+++ b/pylint/extensions/check_elif.py
@@ -31,7 +31,7 @@ class ElseifUsedChecker(BaseTokenChecker):
     name = "else_if_used"
     msgs = {
         "R5501": (
-            'Consider using "elif" instead of "else", replace the "else" by an "elif"'
+            'Consider using "elif" instead of "else", replace the "else" by an "elif",'
             ' then de-indent the code and switch from "if" to "elif" inside the else',
             "else-if-used",
             "Used when an else statement is immediately followed by "

--- a/tests/functional/ext/check_elif/check_elif.txt
+++ b/tests/functional/ext/check_elif/check_elif.txt
@@ -1,4 +1,4 @@
-else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH

--- a/tests/functional/ext/check_elif/check_elif.txt
+++ b/tests/functional/ext/check_elif/check_elif.txt
@@ -1,4 +1,4 @@
-else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else if""":HIGH
-else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else if""":HIGH
-else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else if""":HIGH
-else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else if""":HIGH
+else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH

--- a/tests/functional/ext/check_elif/check_elif.txt
+++ b/tests/functional/ext/check_elif/check_elif.txt
@@ -1,4 +1,4 @@
-else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"" then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", replace the ""else"" by an ""elif"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH

--- a/tests/functional/ext/check_elif/check_elif.txt
+++ b/tests/functional/ext/check_elif/check_elif.txt
@@ -1,4 +1,4 @@
-else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
-else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"", remove the ""else"", then de-indent the code and switch from ""if"" to ""elif"" inside the else":HIGH
+else-if-used:13:8:30:25:my_function:"Consider using ""elif"" instead of ""else"" then ""if"" to remove one indentation level":HIGH
+else-if-used:25:20:26:28:my_function:"Consider using ""elif"" instead of ""else"" then ""if"" to remove one indentation level":HIGH
+else-if-used:44:8:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"" then ""if"" to remove one indentation level":HIGH
+else-if-used:47:12:48:26:_if_in_fstring_comprehension_with_elif:"Consider using ""elif"" instead of ""else"" then ""if"" to remove one indentation level":HIGH

--- a/tests/functional/n/no/no_else_break.txt
+++ b/tests/functional/n/no/no_else_break.txt
@@ -1,7 +1,7 @@
-no-else-break:8:8:11:17:foo1:"Unnecessary ""else"" after ""break""":UNDEFINED
-no-else-break:16:8:21:17:foo2:"Unnecessary ""elif"" after ""break""":UNDEFINED
-no-else-break:28:12:33:21:foo3:"Unnecessary ""else"" after ""break""":UNDEFINED
-no-else-break:41:8:48:17:foo4:"Unnecessary ""else"" after ""break""":UNDEFINED
-no-else-break:54:8:63:17:foo5:"Unnecessary ""elif"" after ""break""":UNDEFINED
-no-else-break:70:12:74:21:foo6:"Unnecessary ""else"" after ""break""":UNDEFINED
-no-else-break:110:8:116:21:bar4:"Unnecessary ""else"" after ""break""":UNDEFINED
+no-else-break:8:8:11:17:foo1:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:16:8:21:17:foo2:"Unnecessary ""elif"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:28:12:33:21:foo3:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:41:8:48:17:foo4:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:54:8:63:17:foo5:"Unnecessary ""elif"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:70:12:74:21:foo6:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:110:8:116:21:bar4:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_break.txt
+++ b/tests/functional/n/no/no_else_break.txt
@@ -1,7 +1,7 @@
-no-else-break:8:8:11:17:foo1:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:16:8:21:17:foo2:"Unnecessary ""elif"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:28:12:33:21:foo3:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:41:8:48:17:foo4:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:54:8:63:17:foo5:"Unnecessary ""elif"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:70:12:74:21:foo6:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:110:8:116:21:bar4:"Unnecessary ""else"" after ""break"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:8:8:11:17:foo1:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:16:8:21:17:foo2:"Unnecessary ""elif"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:28:12:33:21:foo3:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:41:8:48:17:foo4:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:54:8:63:17:foo5:"Unnecessary ""elif"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:70:12:74:21:foo6:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:110:8:116:21:bar4:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_break.txt
+++ b/tests/functional/n/no/no_else_break.txt
@@ -1,7 +1,7 @@
-no-else-break:8:8:11:17:foo1:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:16:8:21:17:foo2:"Unnecessary ""elif"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:28:12:33:21:foo3:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:41:8:48:17:foo4:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:54:8:63:17:foo5:"Unnecessary ""elif"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:70:12:74:21:foo6:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-break:110:8:116:21:bar4:"Unnecessary ""else"" after ""break"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-break:8:8:11:17:foo1:"Unnecessary ""else"" after ""break"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-break:16:8:21:17:foo2:"Unnecessary ""elif"" after ""break"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-break:28:12:33:21:foo3:"Unnecessary ""else"" after ""break"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-break:41:8:48:17:foo4:"Unnecessary ""else"" after ""break"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-break:54:8:63:17:foo5:"Unnecessary ""elif"" after ""break"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-break:70:12:74:21:foo6:"Unnecessary ""else"" after ""break"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-break:110:8:116:21:bar4:"Unnecessary ""else"" after ""break"", remove the ""else"" and de-indent the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_continue.txt
+++ b/tests/functional/n/no/no_else_continue.txt
@@ -1,7 +1,7 @@
-no-else-continue:8:8:11:17:foo1:"Unnecessary ""else"" after ""continue""":UNDEFINED
-no-else-continue:16:8:21:17:foo2:"Unnecessary ""elif"" after ""continue""":UNDEFINED
-no-else-continue:28:12:33:24:foo3:"Unnecessary ""else"" after ""continue""":UNDEFINED
-no-else-continue:41:8:48:17:foo4:"Unnecessary ""else"" after ""continue""":UNDEFINED
-no-else-continue:54:8:63:17:foo5:"Unnecessary ""elif"" after ""continue""":UNDEFINED
-no-else-continue:70:12:74:21:foo6:"Unnecessary ""else"" after ""continue""":UNDEFINED
-no-else-continue:110:8:116:24:bar4:"Unnecessary ""else"" after ""continue""":UNDEFINED
+no-else-continue:8:8:11:17:foo1:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:16:8:21:17:foo2:"Unnecessary ""elif"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:28:12:33:24:foo3:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:41:8:48:17:foo4:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:54:8:63:17:foo5:"Unnecessary ""elif"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:70:12:74:21:foo6:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:110:8:116:24:bar4:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_continue.txt
+++ b/tests/functional/n/no/no_else_continue.txt
@@ -1,7 +1,7 @@
-no-else-continue:8:8:11:17:foo1:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:16:8:21:17:foo2:"Unnecessary ""elif"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:28:12:33:24:foo3:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:41:8:48:17:foo4:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:54:8:63:17:foo5:"Unnecessary ""elif"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:70:12:74:21:foo6:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:110:8:116:24:bar4:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:8:8:11:17:foo1:"Unnecessary ""else"" after ""continue"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-continue:16:8:21:17:foo2:"Unnecessary ""elif"" after ""continue"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-continue:28:12:33:24:foo3:"Unnecessary ""else"" after ""continue"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-continue:41:8:48:17:foo4:"Unnecessary ""else"" after ""continue"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-continue:54:8:63:17:foo5:"Unnecessary ""elif"" after ""continue"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-continue:70:12:74:21:foo6:"Unnecessary ""else"" after ""continue"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-continue:110:8:116:24:bar4:"Unnecessary ""else"" after ""continue"", remove the ""else"" and de-indent the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_continue.txt
+++ b/tests/functional/n/no/no_else_continue.txt
@@ -1,7 +1,7 @@
-no-else-continue:8:8:11:17:foo1:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:16:8:21:17:foo2:"Unnecessary ""elif"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:28:12:33:24:foo3:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:41:8:48:17:foo4:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:54:8:63:17:foo5:"Unnecessary ""elif"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:70:12:74:21:foo6:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-continue:110:8:116:24:bar4:"Unnecessary ""else"" after ""continue"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:8:8:11:17:foo1:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:16:8:21:17:foo2:"Unnecessary ""elif"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:28:12:33:24:foo3:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:41:8:48:17:foo4:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:54:8:63:17:foo5:"Unnecessary ""elif"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:70:12:74:21:foo6:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-continue:110:8:116:24:bar4:"Unnecessary ""else"" after ""continue"", remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_raise.txt
+++ b/tests/functional/n/no/no_else_raise.txt
@@ -1,7 +1,7 @@
-no-else-raise:6:4:11:26:foo1:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:15:4:23:26:foo2:"Unnecessary ""elif"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:29:8:34:30:foo3:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:41:4:48:13:foo4:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:53:4:62:13:foo5:"Unnecessary ""elif"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:68:8:72:17:foo6:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:104:4:110:33:bar4:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:6:4:11:26:foo1:"Unnecessary ""else"" after ""raise"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-raise:15:4:23:26:foo2:"Unnecessary ""elif"" after ""raise"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-raise:29:8:34:30:foo3:"Unnecessary ""else"" after ""raise"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-raise:41:4:48:13:foo4:"Unnecessary ""else"" after ""raise"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-raise:53:4:62:13:foo5:"Unnecessary ""elif"" after ""raise"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-raise:68:8:72:17:foo6:"Unnecessary ""else"" after ""raise"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-raise:104:4:110:33:bar4:"Unnecessary ""else"" after ""raise"", remove the ""else"" and de-indent the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_raise.txt
+++ b/tests/functional/n/no/no_else_raise.txt
@@ -1,7 +1,7 @@
-no-else-raise:6:4:11:26:foo1:"Unnecessary ""else"" after ""raise""":UNDEFINED
-no-else-raise:15:4:23:26:foo2:"Unnecessary ""elif"" after ""raise""":UNDEFINED
-no-else-raise:29:8:34:30:foo3:"Unnecessary ""else"" after ""raise""":UNDEFINED
-no-else-raise:41:4:48:13:foo4:"Unnecessary ""else"" after ""raise""":UNDEFINED
-no-else-raise:53:4:62:13:foo5:"Unnecessary ""elif"" after ""raise""":UNDEFINED
-no-else-raise:68:8:72:17:foo6:"Unnecessary ""else"" after ""raise""":UNDEFINED
-no-else-raise:104:4:110:33:bar4:"Unnecessary ""else"" after ""raise""":UNDEFINED
+no-else-raise:6:4:11:26:foo1:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:15:4:23:26:foo2:"Unnecessary ""elif"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:29:8:34:30:foo3:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:41:4:48:13:foo4:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:53:4:62:13:foo5:"Unnecessary ""elif"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:68:8:72:17:foo6:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:104:4:110:33:bar4:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_raise.txt
+++ b/tests/functional/n/no/no_else_raise.txt
@@ -1,7 +1,7 @@
-no-else-raise:6:4:11:26:foo1:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:15:4:23:26:foo2:"Unnecessary ""elif"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:29:8:34:30:foo3:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:41:4:48:13:foo4:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:53:4:62:13:foo5:"Unnecessary ""elif"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:68:8:72:17:foo6:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-raise:104:4:110:33:bar4:"Unnecessary ""else"" after ""raise"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:6:4:11:26:foo1:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:15:4:23:26:foo2:"Unnecessary ""elif"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:29:8:34:30:foo3:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:41:4:48:13:foo4:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:53:4:62:13:foo5:"Unnecessary ""elif"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:68:8:72:17:foo6:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-raise:104:4:110:33:bar4:"Unnecessary ""else"" after ""raise"", remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_return.txt
+++ b/tests/functional/n/no/no_else_return.txt
@@ -1,7 +1,7 @@
-no-else-return:6:4:11:16:foo1:"Unnecessary ""else"" after ""return""":UNDEFINED
-no-else-return:15:4:23:16:foo2:"Unnecessary ""elif"" after ""return""":UNDEFINED
-no-else-return:29:8:34:20:foo3:"Unnecessary ""else"" after ""return""":UNDEFINED
-no-else-return:41:4:48:13:foo4:"Unnecessary ""else"" after ""return""":UNDEFINED
-no-else-return:53:4:62:13:foo5:"Unnecessary ""elif"" after ""return""":UNDEFINED
-no-else-return:68:8:72:17:foo6:"Unnecessary ""else"" after ""return""":UNDEFINED
-no-else-return:104:4:110:23:bar4:"Unnecessary ""else"" after ""return""":UNDEFINED
+no-else-return:6:4:11:16:foo1:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:15:4:23:16:foo2:"Unnecessary ""elif"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:29:8:34:20:foo3:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:41:4:48:13:foo4:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:53:4:62:13:foo5:"Unnecessary ""elif"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:68:8:72:17:foo6:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:104:4:110:23:bar4:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_return.txt
+++ b/tests/functional/n/no/no_else_return.txt
@@ -1,7 +1,7 @@
-no-else-return:6:4:11:16:foo1:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:15:4:23:16:foo2:"Unnecessary ""elif"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:29:8:34:20:foo3:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:41:4:48:13:foo4:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:53:4:62:13:foo5:"Unnecessary ""elif"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:68:8:72:17:foo6:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:104:4:110:23:bar4:"Unnecessary ""else"" after ""return"" remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:6:4:11:16:foo1:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:15:4:23:16:foo2:"Unnecessary ""elif"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:29:8:34:20:foo3:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:41:4:48:13:foo4:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:53:4:62:13:foo5:"Unnecessary ""elif"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:68:8:72:17:foo6:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:104:4:110:23:bar4:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED

--- a/tests/functional/n/no/no_else_return.txt
+++ b/tests/functional/n/no/no_else_return.txt
@@ -1,7 +1,7 @@
-no-else-return:6:4:11:16:foo1:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:15:4:23:16:foo2:"Unnecessary ""elif"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:29:8:34:20:foo3:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:41:4:48:13:foo4:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:53:4:62:13:foo5:"Unnecessary ""elif"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:68:8:72:17:foo6:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
-no-else-return:104:4:110:23:bar4:"Unnecessary ""else"" after ""return"", remove it and de-indent all the code inside it":UNDEFINED
+no-else-return:6:4:11:16:foo1:"Unnecessary ""else"" after ""return"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-return:15:4:23:16:foo2:"Unnecessary ""elif"" after ""return"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-return:29:8:34:20:foo3:"Unnecessary ""else"" after ""return"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-return:41:4:48:13:foo4:"Unnecessary ""else"" after ""return"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-return:53:4:62:13:foo5:"Unnecessary ""elif"" after ""return"", remove the leading ""el"" from ""elif""":UNDEFINED
+no-else-return:68:8:72:17:foo6:"Unnecessary ""else"" after ""return"", remove the ""else"" and de-indent the code inside it":UNDEFINED
+no-else-return:104:4:110:23:bar4:"Unnecessary ""else"" after ""return"", remove the ""else"" and de-indent the code inside it":UNDEFINED

--- a/tests/functional/u/useless/useless_else_on_loop.txt
+++ b/tests/functional/u/useless/useless_else_on_loop.txt
@@ -1,6 +1,6 @@
-useless-else-on-loop:10:4:11:31:test_return_for:Else clause on loop without a break statement:UNDEFINED
-useless-else-on-loop:18:4:19:31:test_return_while:Else clause on loop without a break statement:UNDEFINED
-useless-else-on-loop:28:0:29:21::Else clause on loop without a break statement:UNDEFINED
-useless-else-on-loop:35:0:36:21::Else clause on loop without a break statement:UNDEFINED
-useless-else-on-loop:40:0:43:13::Else clause on loop without a break statement:UNDEFINED
-useless-else-on-loop:87:4:88:19:test_break_in_orelse_deep2:Else clause on loop without a break statement:UNDEFINED
+useless-else-on-loop:10:4:11:31:test_return_for:Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
+useless-else-on-loop:18:4:19:31:test_return_while:Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
+useless-else-on-loop:28:0:29:21::Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
+useless-else-on-loop:35:0:36:21::Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
+useless-else-on-loop:40:0:43:13::Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
+useless-else-on-loop:87:4:88:19:test_break_in_orelse_deep2:Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED

--- a/tests/functional/u/useless/useless_else_on_loop.txt
+++ b/tests/functional/u/useless/useless_else_on_loop.txt
@@ -1,6 +1,6 @@
-useless-else-on-loop:10:4:11:31:test_return_for:Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
-useless-else-on-loop:18:4:19:31:test_return_while:Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
-useless-else-on-loop:28:0:29:21::Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
-useless-else-on-loop:35:0:36:21::Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
-useless-else-on-loop:40:0:43:13::Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
-useless-else-on-loop:87:4:88:19:test_break_in_orelse_deep2:Else clause on loop without a break statement, remove the else andde-indent all the code inside it:UNDEFINED
+useless-else-on-loop:10:4:11:31:test_return_for:Else clause on loop without a break statement, remove the else and de-indent all the code inside it:UNDEFINED
+useless-else-on-loop:18:4:19:31:test_return_while:Else clause on loop without a break statement, remove the else and de-indent all the code inside it:UNDEFINED
+useless-else-on-loop:28:0:29:21::Else clause on loop without a break statement, remove the else and de-indent all the code inside it:UNDEFINED
+useless-else-on-loop:35:0:36:21::Else clause on loop without a break statement, remove the else and de-indent all the code inside it:UNDEFINED
+useless-else-on-loop:40:0:43:13::Else clause on loop without a break statement, remove the else and de-indent all the code inside it:UNDEFINED
+useless-else-on-loop:87:4:88:19:test_break_in_orelse_deep2:Else clause on loop without a break statement, remove the else and de-indent all the code inside it:UNDEFINED

